test/include/dlaf_test/matrix/util_tile.h

```diff
@@ -265,8 +265,10 @@ void checkNear(ElementGetter&& expected, const Tile<const T, D>& tile, BaseType<
   };
   internal::check(expected, tile, comp, err_message, file, line);
 }
+DLAF_MAKE_CALLABLE_OBJECT(checkNear);
+
 #define CHECK_TILE_NEAR(expected, tile, rel_err, abs_err) \
-  ::dlaf::matrix::test::checkNear(expected, tile, rel_err, abs_err, __FILE__, __LINE__)
+  ::dlaf::matrix::test::checkNear_o(expected, tile, rel_err, abs_err, __FILE__, __LINE__)
```

Comment:
Do you have an example of the compilation failure? This is not really what `DLAF_MAKE_CALLABLE_OBJECT` should be used (more importantly, needed) for. I don't understand why this would fix anything :/

Reply:
Yes, this was commented out a few commits back because of a compile-time error : https://github.com/eth-cscs/DLA-Future/pull/436/files#diff-b5a15fd19989a0843c5344662f5ce7e61e8f2110d28e7ff6ab785aca63bcc44dR40-R65

I used it because the issue appeared to be related to a compile-time failure in the templated and overloaded `checkNear()` function.

Reply:
Do you have an error message? Why does this fix it? `DLAF_MAKE_CALLABLE_OBJECT` _should_ create a transparent wrapper that ends up calling `checkNear` with the same arguments and _should_  have the same compilation failure. If it doesn't, there may be a bug in the wrapper as well...

Reply:
Yes, it's easy to reproduce, just insert the following:

```
TYPED_TEST(MatrixLocalTest, Dummy) {    
  constexpr auto error = TypeUtilities<TypeParam>::error;    
  auto tile = createTile<TypeParam>([](TileElementIndex idx) { return idx.row() + idx.col(); },    
                                    TileElementSize(5, 5), 5);    
  CHECK_TILE_NEAR(tile, tile, error, error);    
}  
```

at the bottom of `test_matrix_local.cpp` and remove the `_o` from checkNear in the macro.

Error message: https://pastebin.com/ebjezAnR